core/node/rpc/add_event.go

```diff
@@ -77,11 +77,20 @@ func (s *Service) addParsedEvent(
 		return err
 	}
 
-	if chainAuthArgs != nil {
-		isEntitled, err := s.chainAuth.IsEntitled(ctx, s.config, chainAuthArgs)
-		if err != nil {
-			return err
+	if len(chainAuthArgsList) > 0 {
+		isEntitled := false
+		var err error
```

Comment:
This looks good. Random nit - hard to see on mobile, is there a reason you predicated this var err here? Looks like you return it immediately if it gets set

Reply:
*pre-declared